### ISSUE TITLE
KSP + NAPT creating stack overflow when used together

### ISSUE THREAD

**Issue by trevjonez** [google/ksp#1321]
Reproduction of the issue is here.
https://github.com/sergei-lapin/napt/pull/19

Adding a touch of println output in the looping code path seems to suggest the issue is likely best fixed within KSP.
![image](https://user-images.githubusercontent.com/5386503/218168136-8bc3eb36-dfd7-43cc-879e-94a4b3e9e305.png)

![image](https://user-images.githubusercontent.com/5386503/218168092-e25c067c-bc68-41d2-b499-54949975a45d.png)

**Comment by neetopia** [comment]
can you give me some context on how NAPT works?

**Comment by trevjonez** [comment]
The TLDR of it is it allows you to avoid the need to generate java stubs like KAPT. Skimming the code it seems to generate a bit of junk java just to trigger the java compile tasks where you can then run annotation processors over the .class outputs of the kotlin compile task? 

Key tradeoff is that you can't reference generated outputs in kotlin code unless you wrap it in java. A mostly trivial impact for vanilla dagger users.

On slack @ZacSweers informed me of potential cache and other incremental issues that NAPT can introduce. I need to dig into that further to confirm real world impact still. But given my main project I suspect going back to kapt will be a no go still.

![image](https://user-images.githubusercontent.com/5386503/218224096-be7a3a91-f5cc-49ab-b2fa-cac6acdeda1e.png)

**Comment by trevjonez** [comment]
Ended up just reverting back to KAPT for now. Once dagger ships KSP support kapt will be gone anyway.
